perfkitbenchmarker/providers/azure/azure_disk.py

```diff
@@ -44,21 +78,37 @@ class AzureDisk(disk.BaseDisk):
   _lock = threading.Lock()
   num_disks = {}
 
-  def __init__(self, disk_spec, vm_name):
+  def __init__(self, disk_spec, vm_name, machine_type):
     super(AzureDisk, self).__init__(disk_spec)
     self.host_caching = FLAGS.azure_host_caching
     self.name = None
     self.vm_name = vm_name
     self.lun = None
 
+    if self.disk_type == PREMIUM_STORAGE:
+      self.metadata = PREMIUM_STORAGE_METADATA
+    elif self.disk_type == STANDARD_DISK:
+      self.metadata = {
```

Comment:
Azure's standard-disk can't be a constant dictionary because you need to look at FLAGS.azure_storage_type to find the replication strategy, and the local SSD metadata can't be a constant dictionary because you need to look at the machine type to find the media. I factored out premium storage because that actually is constant.